File: MultiLanguageSandbox/src/test/deps/cpp/26.cpp
#include <string>
#include <algorithm>
#include <cassert>
#include <cmath>
#include <cstdio>
#include <cstring>
/**
 * Classify the integer x based on the following method:
 * If x is a single-digit number, x belongs to its own class.
 * Otherwise, sum the digits of x, get a new x, and iterate until 
 * the class is determined.
 *
 * Parameters:
 * - x (const char*): The integer to be classified in string form.
 *
 * Returns:
 * int: The class to which the integer x belongs.
 *
 * Usage:
 * >>> classify_integer("24")
 * 6
 */
int classify_integer(const char *x) {
    std::string num_str(x);
    int num = 0;
    
    // Convert the string to an integer
    for (char c : num_str) {
        if (c < '0' || c > '9') {
            // Handle non-digit characters if necessary, though problem says it's an integer string
            return -1; // or handle error appropriately
        }
        num = num * 10 + (c - '0');
    }
    
    // Process the number until it's a single digit
    while (num >= 10) {
        int sum = 0;
        int temp = num;
        while (temp > 0) {
            sum += temp % 10;
            temp /= 10;
        }
        num = sum;
    }
    
    return num;
}
int main() {
    assert(classify_integer("24") == 6);
    assert(classify_integer("39") == 3);
    assert(classify_integer("123456789") == 9);
    assert(classify_integer("123456789012345678901234567890") == 9);
    assert(classify_integer("12345") == 6);
    assert(classify_integer("999999999") == 9);
    // std::printf("All tests passed successfully!\n");
    return 0;
}

File: MultiLanguageSandbox/src/test/deps/pascal/26.pas
{$ASSERTIONS ON}

uses
  SysUtils, StrUtils;
(*
  Classify the integer x as follows:
    - If x is a single-digit number, x belongs to its own class.
    - Otherwise, sum the digits of x, get a new x, and continue this process iteratively 
      until the class is determined.

  @param x: The string representation of the integer to be classified.
  @return: The class to which the integer x belongs.

  Examples:
      >>> classify_integer('24')
      6
*)

var

function classify_integer(x: string): Integer;
var
  num: Integer;
  digit: Integer;
  sum: Integer;
begin
  // Convert the string to an integer
  num := StrToInt(x);
  
  // Handle the case where the number is 0
  if num = 0 then
  begin
    classify_integer := 0;
    Exit;
  end;
  
  // Process the number until it becomes a single digit
  while num >= 10 do
  begin
    sum := 0;
    // Sum all digits of the current number
    while num > 0 do
    begin
      digit := num mod 10;
      sum := sum + digit;
      num := num div 10;
    end;
    num := sum;
  end;
  
  classify_integer := num;
end;
procedure testClassifyInteger;
begin
  Assert(classify_integer('24') = 6);
  Assert(classify_integer('39') = 3);
  Assert(classify_integer('123456789') = 9);
  Assert(classify_integer('123456789012345678901234567890') = 9);
  Assert(classify_integer('12345') = 6);
  Assert(classify_integer('999999999') = 9);
end;

begin
  testClassifyInteger;
end.